src/metabase/query_processor/middleware/desugar.clj

```diff
@@ -27,25 +27,28 @@
     [:time-interval field :next    unit options] (recur [:time-interval field  1 unit options])
 
     [:time-interval field (n :guard #{-1}) unit (_ :guard :include-current)]
-    [:between [:datetime-field field unit] [:relative-datetime n unit] [:relative-datetime 0 unit]]
+    [:between-rhs-ex [:datetime-field field :default] [:relative-datetime n unit] [:relative-datetime 1 unit]]
```

Comment:
Why `:default` bucketing?

Reply:
Using anything else can introduce additional function calls on the `datetime-field` in the resulting SQL which in turn doesn't allow us to utilise index on that field. The idea was that `:relative-datetime` would always return full datetime as well but it would cut out the non-relevant parts (for example `DATE_FORMAT(DATE_ADD(NOW(), INTERVAL -2 MONTH), '%Y-%m-01 00:00:00')` which results in `2019-03-01 00:00:00` for `[:relative-datetime -2 :month]`). This is currently not the case though because the default `:relative-datetime`'s behaviour is to _truncate_ the date without zero-padding it. (this is the bug I'm talking about in another comment) So I opted to introducing another `:relative-datetime-padded` clause :) And now I really don't like how much new stuff this introduces. If you have some ideas please share them with me, I will probably have time to implement them. My main objective here is to reduce, but preferably eliminate, any use of SQL functions around date time fields.